federation/pkg/federation-controller/sync/controller.go

```diff
@@ -336,24 +340,54 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 	if err != nil {
 		glog.Errorf("Failed to get cluster list: %v", err)
 		s.deliver(namespacedName, s.clusterAvailableDelay, false)
-		return
+		return err, reconcileResult
 	}
 
 	operations := make([]util.FederatedOperation, 0)
 	for _, cluster := range clusters {
+		glog.V(3).Infof("Processing cluster: %s for %s: %s", cluster.Name, kind, key)
+
 		clusterObj, found, err := s.informer.GetTargetStore().GetByKey(cluster.Name, key)
 		if err != nil {
 			glog.Errorf("Failed to get %s from %s: %v", key, cluster.Name, err)
 			s.deliver(namespacedName, 0, true)
-			return
+			return err, reconcileResult
 		}
+		reconcileResult[cluster.Name] = found
```

Comment:
Do you really need this?
It seems that irrespective of what you set here, you always override the value based on send (set it to true if send is true and false otherwise)